source/rock/middle/TypeDecl.ooc

```diff
@@ -1253,6 +1253,29 @@ TypeDecl: abstract class extends Declaration {
                     varAcc := VariableAccess new("this", access token)
                     varAcc reverseExpr = access
                     access expr = varAcc
+                    if (res params explicitThis) {
+                        mustBeExplicit := true
+
+                        // Generic variables do not require explicit this
+                        if (vDecl getType()) {
+                            if (vDecl getType() toString() == "Class") {
+                                mustBeExplicit = false
```

Comment:
Filter out variables like `T` from generic classes since the variable is also a type used to declare other variables.